### ISSUE TITLE
"Remove unsorted channels" doesn't work

### ISSUE THREAD

**Issue by UnicornDuckEli5** [PredatH0r/ChanSort#100]
For samsung list after choosing "Remove unsorted channels" I can't remove channels completely with 2017-11-30 version. It keeps adding removed channels at the end of list so there is no point in choosing this option because both work as "Append unsorted channels the end of the list" 2017-11-16 version works fine. I know I can just use which version works for me but if new feature or fix gets added I can't use newest version then

**Comment by Luckysnooze** [comment]
I have the same problem. It would be nice if it could be fixed with one of the next versions.
Thanks for the usefull program.

**Comment by Dyna-mike** [comment]
I have the same problem with version 2017-11-30. Now i have > 1.200 channels on my TV :-(
PS:
Anyway a helpful programm.

**Comment by mody** [comment]
Until I've found this issue, I thought I was doing something wrong.  Happens for me too, version 2017-11-30 and Samsung TV 32C6500.

**Comment by f-alex** [comment]
For me also. Removing on 2017-11-30 don't work for Samsung J channels. But 2017-11-16 work well

**Comment by Nukethepeace** [comment]
Too bad I'm having the exact same issue here on a Samsung UE32H5570

**Comment by DocSneider** [comment]
I also had the same issue here on a Panasonic tx-l42ew5.

**I solved it by using an older version [ChanSort_2015-11-28]**. 

With that version, all unsorted channels where removed when using the option "delete unsorted channels from the list".

**Comment by devau** [comment]
I also have this issue here for Samsung TV Series E. I'm using Chansort 2017-11-30.
So this issue seems to be well known for nearly one year now. Who is going to fix it?
Or is there any workaround?

**Comment by hurda** [comment]
Is this fixed in the recent ChanSort-versions? 2019-02-06

**Comment by PredatH0r** [comment]
It's not fixed yet. I am currently looking into it.
I have to figure out whether it is even possible to remove channels entirely. Some TVs will automatically add the channels back into the list, because it tries to automatically add "new" channels.

**Comment by PredatH0r** [comment]
This is now fixed in version 2019-02-10

**Comment by nemesis63** [comment]
No it's not fixed for me.

**Comment by ghodtyr3** [comment]
This issue not fixed yet @PredatH0r

**Comment by PredatH0r** [comment]
Unfortunately all the various file formats have different requirements on how to "delete" a channel, which makes this a bit difficult for me to implement it in a consistent way.
Some formats require that a channel record is removed from the file, other formats require that the record is marked with a "deleted" flag and some dummy channel number, and some require a valid channel number in addition to the "deleted" flag (which means i have to somehow keep these channels in the list but hide them from the user).
The safest way is to not delete anything, mark the channels as hidden and append them at the end, and that's what I tried to do after receiving several notes that after deleting a channel, the TV would not work properly with the list. But not all TVs have a "hidden" flag, so they just appeared at the end of the list again (e.g. for Samsung .zip lists).

Anyway... In my current work-in-progress version I am trying to clean this mess up. But this will still take some time

**Comment by PredatH0r** [comment]
The new ChanSort version 2019-11-08 has new handling for unsorted/deleted channels across all TV file formats.
For Samsung SCM files the channels are marked as deleted in the file.
For Samsung ZIP files the channels are removed from the file (because there is no marker for "deleted". This may cause that the TV will automatically re-add the channel when it thinks it found a new one)
For Panasonic lists the channel will also be removed from the file.
